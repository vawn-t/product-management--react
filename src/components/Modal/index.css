.modal {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-dialog {
  position: fixed;
  position: relative;
  background: white;
  width: 60%;
  height: auto;
  padding: 2rem;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.button-group {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  text-transform: capitalize;
}

.close-icon {
  position: absolute;
  cursor: pointer;
  right: 1rem;
  top: 1rem;
}
