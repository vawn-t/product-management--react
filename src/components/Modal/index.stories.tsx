// Library
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Component
import { Modal } from '@components/index';

// Styles
import '@assets/styles/App.css';
import '@assets/styles/variables.css';
import './index.css';

export default {
  title: 'Components/Modal',
  component: Modal
} as ComponentMeta<typeof Modal>;

const Template: ComponentStory<typeof Modal> = (args) => <Modal {...args} />;

export const Default = Template.bind({});
Default.args = {
  isVisible: true,
  children: <p>Modal </p>
};
