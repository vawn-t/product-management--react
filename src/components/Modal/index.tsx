// Libraries
import React, { ReactNode } from 'react';

// Components
import { Image } from '@components/index';

// Assets
import { closeIcon } from '@assets/index';

// Styles
import './index.css';

export interface ModalProps {
  children: ReactNode;
  isVisible: boolean;
  onClose: () => void;
}

const Modal: React.FC<ModalProps> = ({ children, isVisible, onClose }) =>
  isVisible ? (
    <div className='modal'>
      <div className='modal-dialog'>
        <div className='close-icon'>
          <Image alt='close icon' imageUrl={closeIcon} onClick={onClose} />
        </div>
        <div className='modal-body'>{children}</div>
      </div>
    </div>
  ) : null;

export default Modal;
