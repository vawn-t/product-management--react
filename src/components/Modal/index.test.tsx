// Libraries
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { screen } from '@testing-library/dom';
import { create } from 'react-test-renderer';

// Components
import Modal, { ModalProps } from './index';

describe('Modal component', () => {
  const defaultProps: ModalProps = {
    children: '',
    isVisible: true,
    onClose: jest.fn()
  };

  let container: HTMLElement;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    container.remove();
    jest.clearAllMocks();
    cleanup();
  });

  test('should render correctly', () => {
    const defaultTree = create(<Modal {...defaultProps} />).toJSON();
    expect(defaultTree).toMatchSnapshot();
  });

  test('should render invisible', () => {
    render(<Modal {...defaultProps} isVisible={false} />, { container });

    expect(container.childElementCount).toBe(0);
  });

  test('should be click', () => {
    const myMock = jest.fn();

    render(<Modal {...defaultProps} onClose={myMock} />, { container });

    const image: HTMLImageElement = screen.getByRole('img');

    fireEvent.click(image);

    expect(myMock).toHaveBeenCalled();
  });
});
