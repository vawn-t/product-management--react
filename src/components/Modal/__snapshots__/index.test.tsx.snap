// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal-dialog"
  >
    <div
      className="close-icon"
    >
      <img
        alt="close icon"
        className="image"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;
